File: src/0201-0300/283 - Move Zeroes/MoveZeroes.java
public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        int index = 0, end = nums.length;
        for (int i = 0; i < end; i++) {
            if (nums[i] != 0)
                nums[index++] = nums[i];
        }
        while (index < end)
            nums[index++] = 0;
    }
}

File: src/0201-0300/283 - Move Zeroes/move_zeroes.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let (mut index, mut end) = (0, nums.len());

        while index < end {
            if nums[index] == 0 {
                nums.remove(index);
                nums.push(0);
                end -= 1;
            } else {
                index += 1;
            }
        }
    }
}
